### ISSUE TITLE
V8JS Extension

### ISSUE THREAD

**Issue by peterkeen** [heroku/heroku-buildpack-php#80]
Could we please get access to the v8js extension?

http://php.net/manual/en/book.v8js.php

**Comment by cihantas** [comment]
I'd also be pleased if v8js would be included in the official Heroku php buildpack.

**Comment by dzuelke** [comment]
I don't think we will support it anytime soon. You can build and use the extension yourself without running a forked buildpack; see https://github.com/heroku/heroku-buildpack-php/blob/master/README.md#custom-platform-repositories for instructions.

**Comment by hacfi** [comment]
@peterkeen @cihantas Have a look at http://stesie.github.io/2015/11/v8js-buildpack - that might it up & running on Heroku!

**Comment by dzuelke** [comment]
That no longer works, or at least is no longer advisable, @hacfi. The buildpack can now natively use additional repositories with runtimes or extensions, no forking necessary (except for the person who maintains the repository with e.g. a V8JS extension, it's easier to maintain that way, but even that person will run actual applications using the official buildpack).

**Comment by hacfi** [comment]
@dzuelke Thanks for the quick response and the clarification!

**Comment by dzuelke** [comment]
Maybe @stesie can rework #134 to use the new platform repo infrastructure, and then someone can host the extension in an S3 bucket? :)

**Comment by stesie** [comment]
here we go ...

Github repo with the build formulas is at https://github.com/phpv8/heroku-v8js/tree/master

I've just built V8 version 5.0.71.25 and php-v8js itself (in latest release versions, i.e. 0.6.2 for PHP 5 and 1.2.0 for PHP 7).  Just published to a new S3 bucket *heroku-v8js* of my own (let's see how much traffic that will cost) ...

To use it simply use default php buildpack with following extra setting:

```
heroku config:set HEROKU_PHP_PLATFORM_REPOSITORIES="https://heroku-v8js.s3.amazonaws.com/dist-cedar-14-stable/packages.json"
```

**Comment by dzuelke** [comment]
Works beautifully, great job @stesie!

**Comment by dzuelke** [comment]
@stesie can you please allow public listing on that bucket? You can now (see develop branch and  30f97f8) also use `UPSTREAM_S3_BUCKET` and `UPSTREAM_S3_PREFIX`, then you don't have to worry about having those PHP dependencies in your bucket for building.

**Comment by stesie** [comment]
@dzuelke public listing on `heroku-v8js` bucket is now enabled.

**Comment by matiosfree** [comment]
Still have the same issue. phpv8/heroku-v8js  doesn't wotk with heroku-18.
When I run composer update I get: "The request PHP extension ext-v8js * is missing from your system. Install or enable PHP's v8js extension."
